lib/experimental/class-gutenberg-rest-global-styles-revisions-controller.php

```diff
@@ -135,24 +135,24 @@ public function prepare_item_for_response( $item, $request ) {
 		$fields = $this->get_fields_for_response( $request );
 
 		if ( rest_is_field_included( 'author', $fields ) ) {
-			$data['author'] = (int) $parent->post_author;
+			$data['author'] = (int) $item->post_author;
```

Comment:
I have a separate PR for these changes including unit tests: https://github.com/WordPress/gutenberg/pull/50117

Just committing them here to aid testing